File: petclinic-java-gradle-react-tailwind-ts/src/main/java/dev/pollito/petclinic_java_gradle_react_tailwind_ts/owner/OwnerService.java
package dev.pollito.petclinic_java_gradle_react_tailwind_ts.owner;

import java.util.Map;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface OwnerService {

    Page<Owner> findAll(String filter, Pageable pageable);

    Owner get(Integer id);

    Integer create(Owner owner);

    void update(Integer id, Owner owner);

    void delete(Integer id);

    Map<Integer, Integer> getOwnerValues();
}

File: petclinic-kt-gradle-thymeleaf-tailwind-webpack-htmx-ts/src/main/kotlin/dev/pollito/petclinic_kt_gradle_thymeleaf_tailwind_webpack_htmx_ts/owner/OwnerService.kt
package dev.pollito.petclinic_kt_gradle_thymeleaf_tailwind_webpack_htmx_ts.owner

import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable

interface OwnerService {

    fun findAll(filter: String?, pageable: Pageable): Page<Owner>

    fun `get`(id: Int): Owner

    fun create(owner: Owner): Int

    fun update(id: Int, owner: Owner)

    fun delete(id: Int)

    fun getOwnerValues(): Map<Int, Int>
}
